Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewColumn.cs

```diff
@@ -0,0 +1,338 @@
+        // </summary>
+        // <param name="cellTemplate">An existing <see cref="T:System.Windows.Forms.DataGridViewCell"/> to use as a template. </param>
+        // public DataGridViewColumn(DataGridViewCell cellTemplate)
+        
+        // <returns>
+        // An <see cref="T:System.Object"/> that represents the cloned <see cref="T:System.Windows.Forms.DataGridViewBand"/>.
+        // </returns>
+        // <PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        // public override object Clone()
+        
+        // <param name="disposing">true to release both managed and unmanaged resources; false to release only unmanaged resources.</param>
+        // protected override void Dispose(bool disposing)
+        
+        /// <summary>
+        /// Calculates the ideal width of the column based on the specified criteria.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The ideal width, in pixels, of the column.
+        /// </returns>
+        /// <param name="autoSizeColumnMode">A <see cref="T:System.Windows.Forms.DataGridViewAutoSizeColumnMode"/> value that specifies an automatic sizing mode. </param><param name="fixedHeight">true to calculate the width of the column based on the current row heights; false to calculate the width with the expectation that the row heights will be adjusted.</param><exception cref="T:System.ArgumentException"><paramref name="autoSizeColumnMode"/> is <see cref="F:System.Windows.Forms.DataGridViewAutoSizeColumnMode.NotSet"/>, <see cref="F:System.Windows.Forms.DataGridViewAutoSizeColumnMode.None"/>, or <see cref="F:System.Windows.Forms.DataGridViewAutoSizeColumnMode.Fill"/>. </exception><exception cref="T:System.ComponentModel.InvalidEnumArgumentException"><paramref name="autoSizeColumnMode"/> is not a valid <see cref="T:System.Windows.Forms.DataGridViewAutoSizeColumnMode"/> value. </exception>
+        public virtual int GetPreferredWidth(DataGridViewAutoSizeColumnMode autoSizeColumnMode, bool fixedHeight)
+        {
+            Contract.Requires(
+                !(autoSizeColumnMode == DataGridViewAutoSizeColumnMode.NotSet ||
```

Comment:
Certainly, but for the more complex contracts, I've try to stick to just negating the condition that would cause a exception to be thrown. This is kind of the edge of that.